wordpress-mobile/WordPress-iOS
# Fixes magic links.

WPAuth PR: https://github.com/wordpress-mobile/WordPressAuthenticator-iOS/pull/553
WPKit PR: https://github.com/wordpress-mobile/WordPressKit-iOS/pull/322

Fixes a bug that was causing magic links to sometimes not execute the right tracking calls.

## Testing:

Please breakpoints here:

- https://github.com/wordpress-mobile/WordPressAuthenticator-iOS/blob/7099abe30d93fe4f4ab22a02ca1ab6576d1cfa97/WordPressAuthenticator/Authenticator/WordPressAuthenticator.swift#L342
- https://github.com/wordpress-mobile/WordPressAuthenticator-iOS/blob/7099abe30d93fe4f4ab22a02ca1ab6576d1cfa97/WordPressAuthenticator/NUX/NUXLinkAuthViewController.swift#L36


### Test 1: Login -> Signup -> Login

1. Be logged out
2. Request a magic link to login to an existing account of yours.  Don’t tap on the link.
3. Go back and request another magic link to a new account.  (Hint: If you’re using google you can use the + trick and use yourexistinguser+wp2@gmail.com)
4. Now using your device, tap on the link from the first email (the login email from step 2)
5. Make sure both breakpoints take you through the login flow.
6. Make sure the epilogue shown is the login epilogue.


### Test 2: Signup -> Login -> Signup

1. Be logged out
2. Request a magic link to sign up with a new account.  Don’t tap on the link.
3. Go back and request another magic link to an existing account.
4. Now using your device, tap on the link from the first email (the signup email from step 2)
5. Make sure both breakpoints take you through the signup flow.
6. Make sure the epilogue shown is the signup epilogue.

## PR submission checklist:
